### PR TITLE
feat(ui): add code search functionality

### PR DESCRIPTION
**Pull request by liangfung**
## Changes
Add code search functionality to code browser. When searching, it will search the codebase and display the search results based on the repo and branch currently selected by the user.

### Screen record
https://jam.dev/c/117966d1-a7a6-4195-9f96-fc90c72df164

### CODE REVIEW

**Comment by wwayne** [comment]
The shortcut doesn't seem to work (pressing 'S' does not focus on the search field)

Screenshot 2024-07-08 11 37 03

**Comment by wwayne** [comment]
search `react` -> click a link in the result -> browser back -> empty
https://jam.dev/c/0aa4e9e9-36d9-4bb7-99fe-1f512deb9833

**Review by wwayne — CHANGES_REQUESTED** [review]
Code LGTM, but 2 UI bugs, left comments above

File: ee/tabby-ui/app/files/components/code-search-bar.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+      onSubmit={onSubmit}
+      className={cn(
+        'flex w-full shrink-0 items-center bg-background px-4 py-3.5 transition duration-500 ease-in-out',
+        className
+      )}
+    >
+      <div className="relative w-full">
+        <Input
+          ref={inputRef}
+          placeholder="Search repository..."
```
Comment (wsxiaoys) [inline ee/tabby-ui/app/files/components/code-search-bar.tsx:53]:
```suggestion
          placeholder="Search for code..."
```

**Review by wsxiaoys — APPROVED** [review]

**Pull request merged**